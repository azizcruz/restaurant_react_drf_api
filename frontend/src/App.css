.layout {
    background-color: #fff !important;
}