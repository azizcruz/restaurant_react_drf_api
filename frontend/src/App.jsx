import React, { Component } from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom"
import "./App.css";
import "antd/dist/antd.css";
import CustomLayout from "./containers/CustomLayout";
import DataTable from "./components/DataTable";
import FilterInput from "./components/FilterInput";
import Error from "./components/Error"
import axios from "axios";
import { connect } from "react-redux";

class App extends Component {

  fetchData = (filter = false, filterBy = "") => {
    let endpoint = "/api/dishes";
    if (filter) {
      // Send a dispatch to start loading.
      this.props.dispatch({ type: "HANDLE_LOADING", isLoading: true });
      endpoint = `/api/dishes?filter=${filter}&name=${filterBy}`;
    }
    axios
      .get(endpoint)
      .then(res => {
        // Send a dispatch to add the returned data to the single source of data.
        this.props.dispatch({ type: "ADD_DATA", fetched_data: res.data, isLoading: false })
      })
      .catch(err => {
        console.log(err);
      });
  }

  componentWillMount() {
    this.fetchData();
  }
  render() {
    return (



      <div className="App">
        <BrowserRouter>
          <div>
            <CustomLayout>
              {/* fetchData(filter_state, filter_by) */}
              <FilterInput onFilter={value => this.fetchData(true, value)} />
              <Switch>
                <Route path="/" component={DataTable} exact />
                <Route component={ Error } />>
              </Switch>
            </CustomLayout>
          </div>
        </BrowserRouter>
      </div>

    );
  }
}

const mapStateToProps = state => ({
  data: state.data,
  isLoading: state.isLoading
});
export default connect(mapStateToProps)(App);
