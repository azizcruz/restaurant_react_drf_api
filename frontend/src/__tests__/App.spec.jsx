import React from "react";
import { create } from "react-test-renderer";
import App from "../App";

describe("App component", () => {
    test("it matches the snapshot", () => {
      const component = create(<App />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });