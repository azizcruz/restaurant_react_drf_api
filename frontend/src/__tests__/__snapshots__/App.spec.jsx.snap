// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component it matches the snapshot 1`] = `
<div
  className="App"
>
  <section
    className="layout ant-layout"
  >
    <header
      className="ant-layout-header"
    >
      <h1
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Restaurant Menu
      </h1>
    </header>
    <main
      className="ant-layout-content"
      style={
        Object {
          "padding": "0 50px",
        }
      }
    >
      <div
        style={
          Object {
            "minHeight": "84vh",
            "padding": "24px 0",
          }
        }
      >
        <div>
          <div
            className="ant-row"
          >
            <div
              className="ant-col ant-col-sm-12 ant-col-md-6 ant-col-lg-4"
            >
              <span
                className="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
              >
                <span
                  className="ant-input-wrapper ant-input-group"
                >
                  <input
                    className="ant-input"
                    disabled={false}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Filter by name"
                    style={null}
                    type="text"
                    value=""
                  />
                  <span
                    className="ant-input-group-addon"
                  >
                    <button
                      className="ant-btn ant-input-search-button ant-btn-primary"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        aria-label="icon: search"
                        className="anticon anticon-search"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="search"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                          />
                        </svg>
                      </i>
                    </button>
                  </span>
                </span>
              </span>
            </div>
          </div>
          <br />
          <br />
        </div>
        <div>
          <div
            className="ant-table-wrapper"
          >
            <div
              className="ant-spin-nested-loading"
            >
              <div>
                <div
                  className="ant-spin ant-spin-spinning ant-table-without-pagination ant-table-spin-holder"
                >
                  <span
                    className="ant-spin-dot ant-spin-dot-spin"
                  >
                    <i
                      className="ant-spin-dot-item"
                    />
                    <i
                      className="ant-spin-dot-item"
                    />
                    <i
                      className="ant-spin-dot-item"
                    />
                    <i
                      className="ant-spin-dot-item"
                    />
                  </span>
                </div>
              </div>
              <div
                className="ant-spin-container ant-spin-blur"
              >
                <div
                  className="ant-table ant-table-default ant-table-bordered ant-table-empty ant-table-scroll-position-left"
                  style={Object {}}
                >
                  <div
                    className="ant-table-content"
                  >
                    <div
                      className="ant-table-body"
                      onScroll={[Function]}
                      onWheel={[Function]}
                      style={Object {}}
                    >
                      <table
                        className=""
                        style={Object {}}
                      >
                        <colgroup>
                          <col
                            style={
                              Object {
                                "minWidth": undefined,
                                "width": undefined,
                              }
                            }
                          />
                          <col
                            style={
                              Object {
                                "minWidth": undefined,
                                "width": undefined,
                              }
                            }
                          />
                          <col
                            style={
                              Object {
                                "minWidth": undefined,
                                "width": undefined,
                              }
                            }
                          />
                        </colgroup>
                        <thead
                          className="ant-table-thead"
                        >
                          <tr
                            style={
                              Object {
                                "height": null,
                              }
                            }
                          >
                            <th
                              className="ant-table-column-has-actions ant-table-column-has-sorters"
                              onClick={[Function]}
                            >
                              <span
                                className="ant-table-header-column"
                              >
                                <div
                                  className="ant-table-column-sorters"
                                >
                                  <span
                                    className="ant-table-column-title"
                                  >
                                    Name
                                  </span>
                                  <span
                                    className="ant-table-column-sorter"
                                  >
                                    <div
                                      className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                      title="Sort"
                                    >
                                      <i
                                        aria-label="icon: caret-up"
                                        className="anticon anticon-caret-up ant-table-column-sorter-up off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-up"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                      <i
                                        aria-label="icon: caret-down"
                                        className="anticon anticon-caret-down ant-table-column-sorter-down off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-down"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                    </div>
                                  </span>
                                </div>
                              </span>
                            </th>
                            <th
                              className="ant-table-column-has-actions ant-table-column-has-sorters"
                              onClick={[Function]}
                            >
                              <span
                                className="ant-table-header-column"
                              >
                                <div
                                  className="ant-table-column-sorters"
                                >
                                  <span
                                    className="ant-table-column-title"
                                  >
                                    Price
                                  </span>
                                  <span
                                    className="ant-table-column-sorter"
                                  >
                                    <div
                                      className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                      title="Sort"
                                    >
                                      <i
                                        aria-label="icon: caret-up"
                                        className="anticon anticon-caret-up ant-table-column-sorter-up off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-up"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                      <i
                                        aria-label="icon: caret-down"
                                        className="anticon anticon-caret-down ant-table-column-sorter-down off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-down"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                    </div>
                                  </span>
                                </div>
                              </span>
                            </th>
                            <th
                              className="ant-table-column-has-actions ant-table-column-has-sorters"
                              onClick={[Function]}
                            >
                              <span
                                className="ant-table-header-column"
                              >
                                <div
                                  className="ant-table-column-sorters"
                                >
                                  <span
                                    className="ant-table-column-title"
                                  >
                                    Availability
                                  </span>
                                  <span
                                    className="ant-table-column-sorter"
                                  >
                                    <div
                                      className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                      title="Sort"
                                    >
                                      <i
                                        aria-label="icon: caret-up"
                                        className="anticon anticon-caret-up ant-table-column-sorter-up off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-up"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                      <i
                                        aria-label="icon: caret-down"
                                        className="anticon anticon-caret-down ant-table-column-sorter-down off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-down"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                    </div>
                                  </span>
                                </div>
                              </span>
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          className="ant-table-tbody"
                        />
                      </table>
                    </div>
                    <div
                      className="ant-table-placeholder"
                    >
                      <div
                        className="ant-empty ant-empty-normal"
                      >
                        <div
                          className="ant-empty-image"
                        >
                          <img
                            alt="No Data"
                            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                          />
                        </div>
                        <p
                          className="ant-empty-description"
                        >
                          No Data
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      className="ant-layout-footer"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Abdelaziz Abdelioua
    </footer>
  </section>
</div>
`;

exports[`DataTable component it matches the snapshot 1`] = `
<div
  className="App"
>
  <section
    className="layout ant-layout"
  >
    <header
      className="ant-layout-header"
    >
      <h1
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Restaurant Menu
      </h1>
    </header>
    <main
      className="ant-layout-content"
      style={
        Object {
          "padding": "0 50px",
        }
      }
    >
      <div
        style={
          Object {
            "minHeight": "84vh",
            "padding": "24px 0",
          }
        }
      >
        <div>
          <div
            className="ant-row"
          >
            <div
              className="ant-col ant-col-sm-12 ant-col-md-6 ant-col-lg-4"
            >
              <span
                className="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
              >
                <span
                  className="ant-input-wrapper ant-input-group"
                >
                  <input
                    className="ant-input"
                    disabled={false}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Filter by name"
                    style={null}
                    type="text"
                    value=""
                  />
                  <span
                    className="ant-input-group-addon"
                  >
                    <button
                      className="ant-btn ant-input-search-button ant-btn-primary"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        aria-label="icon: search"
                        className="anticon anticon-search"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="search"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                          />
                        </svg>
                      </i>
                    </button>
                  </span>
                </span>
              </span>
            </div>
          </div>
          <br />
          <br />
        </div>
        <div>
          <div
            className="ant-table-wrapper"
          >
            <div
              className="ant-spin-nested-loading"
            >
              <div>
                <div
                  className="ant-spin ant-spin-spinning ant-table-without-pagination ant-table-spin-holder"
                >
                  <span
                    className="ant-spin-dot ant-spin-dot-spin"
                  >
                    <i
                      className="ant-spin-dot-item"
                    />
                    <i
                      className="ant-spin-dot-item"
                    />
                    <i
                      className="ant-spin-dot-item"
                    />
                    <i
                      className="ant-spin-dot-item"
                    />
                  </span>
                </div>
              </div>
              <div
                className="ant-spin-container ant-spin-blur"
              >
                <div
                  className="ant-table ant-table-default ant-table-bordered ant-table-empty ant-table-scroll-position-left"
                  style={Object {}}
                >
                  <div
                    className="ant-table-content"
                  >
                    <div
                      className="ant-table-body"
                      onScroll={[Function]}
                      onWheel={[Function]}
                      style={Object {}}
                    >
                      <table
                        className=""
                        style={Object {}}
                      >
                        <colgroup>
                          <col
                            style={
                              Object {
                                "minWidth": undefined,
                                "width": undefined,
                              }
                            }
                          />
                          <col
                            style={
                              Object {
                                "minWidth": undefined,
                                "width": undefined,
                              }
                            }
                          />
                          <col
                            style={
                              Object {
                                "minWidth": undefined,
                                "width": undefined,
                              }
                            }
                          />
                        </colgroup>
                        <thead
                          className="ant-table-thead"
                        >
                          <tr
                            style={
                              Object {
                                "height": null,
                              }
                            }
                          >
                            <th
                              className="ant-table-column-has-actions ant-table-column-has-sorters"
                              onClick={[Function]}
                            >
                              <span
                                className="ant-table-header-column"
                              >
                                <div
                                  className="ant-table-column-sorters"
                                >
                                  <span
                                    className="ant-table-column-title"
                                  >
                                    Name
                                  </span>
                                  <span
                                    className="ant-table-column-sorter"
                                  >
                                    <div
                                      className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                      title="Sort"
                                    >
                                      <i
                                        aria-label="icon: caret-up"
                                        className="anticon anticon-caret-up ant-table-column-sorter-up off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-up"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                      <i
                                        aria-label="icon: caret-down"
                                        className="anticon anticon-caret-down ant-table-column-sorter-down off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-down"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                    </div>
                                  </span>
                                </div>
                              </span>
                            </th>
                            <th
                              className="ant-table-column-has-actions ant-table-column-has-sorters"
                              onClick={[Function]}
                            >
                              <span
                                className="ant-table-header-column"
                              >
                                <div
                                  className="ant-table-column-sorters"
                                >
                                  <span
                                    className="ant-table-column-title"
                                  >
                                    Price
                                  </span>
                                  <span
                                    className="ant-table-column-sorter"
                                  >
                                    <div
                                      className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                      title="Sort"
                                    >
                                      <i
                                        aria-label="icon: caret-up"
                                        className="anticon anticon-caret-up ant-table-column-sorter-up off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-up"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                      <i
                                        aria-label="icon: caret-down"
                                        className="anticon anticon-caret-down ant-table-column-sorter-down off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-down"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                    </div>
                                  </span>
                                </div>
                              </span>
                            </th>
                            <th
                              className="ant-table-column-has-actions ant-table-column-has-sorters"
                              onClick={[Function]}
                            >
                              <span
                                className="ant-table-header-column"
                              >
                                <div
                                  className="ant-table-column-sorters"
                                >
                                  <span
                                    className="ant-table-column-title"
                                  >
                                    Availability
                                  </span>
                                  <span
                                    className="ant-table-column-sorter"
                                  >
                                    <div
                                      className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                      title="Sort"
                                    >
                                      <i
                                        aria-label="icon: caret-up"
                                        className="anticon anticon-caret-up ant-table-column-sorter-up off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-up"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                      <i
                                        aria-label="icon: caret-down"
                                        className="anticon anticon-caret-down ant-table-column-sorter-down off"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="caret-down"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="0 0 1024 1024"
                                          width="1em"
                                        >
                                          <path
                                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                          />
                                        </svg>
                                      </i>
                                    </div>
                                  </span>
                                </div>
                              </span>
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          className="ant-table-tbody"
                        />
                      </table>
                    </div>
                    <div
                      className="ant-table-placeholder"
                    >
                      <div
                        className="ant-empty ant-empty-normal"
                      >
                        <div
                          className="ant-empty-image"
                        >
                          <img
                            alt="No Data"
                            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                          />
                        </div>
                        <p
                          className="ant-empty-description"
                        >
                          No Data
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      className="ant-layout-footer"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Abdelaziz Abdelioua
    </footer>
  </section>
</div>
`;
